### DISCUSSION TITLE
Hi

### DISCUSSION THREAD

**Discussion by netdudeuk** [open-toontown/open-toontown#61] [Q&A]
Where's the actual PC / Mac client software though ?

Thanks

**Comment by DarthMDev** [comment]
There is no client, you have to run the source code through python

**Comment by netdudeuk** [comment]
There's no concept of running the server code on a remote machine and something else on the PC, which is how TTO worked ?

Thanks

**Comment by EmeraldShift** [comment]
There is a separate server and client program, and the source code for both is in this repo. The setup is roughly like this: On a remote machine, the Astron server (in the `astron/` directory) is running, as well as the "AI server" and "UberDOGs" (the source code for these is all over the place, look for `.py` files that end in "AI" and "UD", respectively). Then, on the PC, you'll run the client, which uses most of the remaining `.py` files.

There are scripts to run each of these pieces in the repo. If you're on Windows, for example, the `win32/*.bat` scripts will start each piece individually. From the readme:
>  make sure you start the following scripts in order:
> `Astron Server -> UberDOG (UD) Server -> AI (District) Server -> Game Client`
